### DISCUSSION TITLE
RGBA support

### DISCUSSION THREAD

**Discussion by Marek68** [baSSiLL/SharpAvi#56] [Q&A] [answered]
I would like to ask you, if is there support for creating AVI video file with alpha channel from raw (RGBA) frames?
I receive raw frames from frame-engine and I need to make uncompress avi (RGBA) with alpha channel in real time.
Thanks a lot!

**Comment by baSSiLL** [comment, marked as answer]
I'm not sure if the alpha channel in AVI files is respected by video players. However, you can give it a try.
Create a video stream using the `AviWriter.AddVideoStream` method - it sets the uncompressed format and 32 bits per pixel by default. And pass your 32-bit bitmaps to the `WriteFrame` method. Remember that in this case (when not using the `UncompressedVideoEncoder` for pre-processing frames) bitmap data should be in the bottom-up layout - the bottom line of a picture goes first, the top line goes last.

**Reply by Marek68** [reply]
Thanks for your advice, it works perfect with alpha channel. Only thing that I had to change is pixel order from RGBA to BGRA for every frame.
